crates/re_tuid/src/lib.rs

```diff
@@ -100,6 +100,19 @@ impl Tuid {
         }
     }
 
+    /// Returns a new `Tuid` with the same timestamp but a different random value.
+    ///
+    /// Beware: wrong usage can easily lead to conflicts.
+    /// Prefer [`Tuid::random`] when unsure.
+    #[inline]
+    pub fn reroll(&self) -> Self {
+        Self {
+            time_ns: self.time_ns,
+            // Leave top bit at zero so we have plenty of room to grow.
+            inc: random_u64() & !(1_u64 << 63),
```

Comment:
This is wrong: we must reroll in the range `self.inc + 1 .. MAX` because intra-timestamp order is actually part of the public API (and is fully defined when logging from a single thread)!

E.g. with the following setup
```python
rr.set_time_sequence("frame", 10)
rr.log("a", rr.Points3D([0, 0, 0]))
rr.log("a", rr.Points3D([])
```
must return `points=[]` when querying for `"a"` at `frame=10`.